test/e2e/resourcepropagation/cronjobs.go

```diff
@@ -30,25 +31,49 @@ import (
 var _ = ginkgo.Describe("CronJob Propagation", func() {
 	f := framework.NewFramework("cronjob-propagation", framework.FrameworkOptions{CreateNamespace: true})
 
-	resourcePropagationTest(
-		f,
-		&resourcePropagationTestConfig[*batchv1b1.CronJob]{
-			gvr:           batchv1b1.SchemeGroupVersion.WithResource("jobs"),
-			objectFactory: resources.GetSimpleCronJob,
-			clientGetter: func(client kubernetes.Interface, namespace string) resourceClient[*batchv1b1.CronJob] {
-				return client.BatchV1beta1().CronJobs(namespace)
+	if framework.CronJobV1Available {
```

Comment:
Must we expose `CronJobV1Available` on the framework? It only seems useful in this test case. Can we query it here?

Reply:
The query needs to be performed during the Run phase. I refactored the generic `resourcePropagationTest` function to not have include any container nodes, to allow it to be conditionally run during the Run phase.

Reply:
The previous implementation of exposing `CronJobV1Available` is also wrong as it will always be false during the Tree Construction phase